p2p/pex/pex_reactor.go

```diff
@@ -372,6 +364,37 @@ func (r *PEXReactor) ensurePeers() {
 	}
 }
 
+func (r *PEXReactor) dialPeer(addr *p2p.NetAddress) {
+	// 1s == (1e9 ns) == (1 Billion ns)
+	billionNs := float64(time.Second.Nanoseconds())
+
+	// exponential backoff if it's not our first attempt to dial given address
+	var attempts int
+	if lAttempts, attempted := r.attemptsToDial.Load(addr.DialString()); attempted {
+		attempts = lAttempts.(int)
+		jitterSeconds := time.Duration(rand.Float64() * billionNs)
```

Comment:
`billionNs` is the same as `float64(time.Second)`, don't think we need that extra variable, but we might benefit from having it configurable.